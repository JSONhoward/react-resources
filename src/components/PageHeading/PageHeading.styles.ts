import styled from 'styled-components';

export const StyledPageHeading = styled('div')`
display: flex;
z-index: 1;

h1 {
    color: ${({theme: {ReactPrimary}}) => ReactPrimary};
    font-size: 4rem;
}
`