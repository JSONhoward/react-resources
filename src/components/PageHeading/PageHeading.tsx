import React from 'react'

import { StyledPageHeading } from './PageHeading.styles';

type Props = {
    heading: string
}

const PageHeading: React.FC<Props> = ({ heading }) => {
    return (
        <StyledPageHeading>
            <h1>{heading}</h1>
        </StyledPageHeading>
    )
}

export default PageHeading
