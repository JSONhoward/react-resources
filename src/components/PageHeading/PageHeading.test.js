import React from 'react'
import {create} from 'react-test-renderer'
import {screen, render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import PageHeading from './'

describe('PageHeading Tests', () => {
    test('Snapshot Test', () => {
        const tree = create(<PageHeading />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('Renders Prop', () => {
        const testHeading = 'test heading'
        render(<PageHeading heading={testHeading} />)

        expect(screen.getByText(testHeading)).toBeInTheDocument()
    })
})