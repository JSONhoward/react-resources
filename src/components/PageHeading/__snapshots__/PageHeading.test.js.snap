// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeading Tests Snapshot Test 1`] = `
<div
  className="sc-bdnylx iWWFrL"
>
  <h1 />
</div>
`;
